macos/FluentUI/Link.swift

```diff
@@ -0,0 +1,127 @@
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+/// A lightweight hyperlink control
+@objc(MSFLink)
+open class Link : NSButton {
+	
+	/// Initializes a hyperlink with a title and an underlying URL that opens when clicked
+	/// - Parameters:
+	///   - title: The visible text of the link that the user sees.
+	///   - url: The URL that is opened when the link is clicked
+	@objc public init(title: String, url: NSURL) {
+		self.url = url
+		super.init(frame: .zero)
+		self.title = title
+		initialize()
+	}
+	
+	/// Initializes a hyperlink with a title and no URL, useful if you plan to override the Target/Action
+	/// - Parameters:
+	///   - title: The visible text of the link that the user sees.
+	@objc public init(title: String) {
+		super.init(frame: .zero)
+		self.title = title
+		initialize()
+	}
+	
+	public override init(frame frameRect: NSRect) {
+		super.init(frame: frameRect)
+		self.title = ""
+		initialize()
+	}
+	
+	@available(*, unavailable)
+	required public init?(coder decoder: NSCoder) {
+		preconditionFailure()
+	}
+	
+	open func initialize() {
+		alignment = .natural
+		isBordered = false
+		target = self
+		action = #selector(linkClicked)
+		updateTitle()
+	}
+	
+	/// The URL that is opened when the link is clicked
+	@objc public var url: NSURL?
+	
+	@objc public var showsUnderlineWhileMouseInside: Bool = false {
+		didSet {
+			guard oldValue != showsUnderlineWhileMouseInside else {
+				return
+			}
+			updateTitle()
+		}
+	}
+	
+	/// The text displayed on the control, stylized to look like a hyperlink
+	open override var title: String {
+		didSet {
+			guard oldValue != title else {
+				return
+			}
+			updateTitle()
+		}
+	}
+	
+	private var trackingArea: NSTrackingArea?
+	
+	override public func updateTrackingAreas() {
+		super.updateTrackingAreas()
+
+		// Remove existing trackingArea
+		if let trackingArea = trackingArea {
+			removeTrackingArea(trackingArea)
+			self.trackingArea = nil
+		}
+
+		// Create a new trackingArea
+		let opts: NSTrackingArea.Options = [.mouseEnteredAndExited, .activeAlways]
+		let trackingArea = NSTrackingArea(
+			rect: bounds,
+			options: opts,
+			owner: self,
+			userInfo: nil
+		)
+		addTrackingArea(trackingArea)
+		self.trackingArea = trackingArea
+	}
+	
+	open override func mouseEntered(with event: NSEvent) {
+		mouseEntered = true
+		updateTitle()
+	}
+
+	open override func mouseExited(with event: NSEvent) {
+		mouseEntered = false
+		updateTitle()
+	}
+
+	private var mouseEntered = false
+	
+	private func updateTitle() {
+		let titleAttributes = (showsUnderlineWhileMouseInside && mouseEntered) ? underlinedlinkAttributes : linkAttributes
+		self.attributedTitle = NSAttributedString(string: title, attributes: titleAttributes)
+	}
+	
+	@objc private func linkClicked() {
+		if let url = url {
+			NSWorkspace.shared.open(url as URL)
+		}
+	}
+}
+
+fileprivate let linkAttributes: [NSAttributedString.Key : Any] = [
+	.foregroundColor : NSColor.linkColor
```

Comment:
very minor. I thought convention in Swift was no space before semi-colon but space after. Not part of this review (and can ignore this comment in this review), but time for a swift format step_

Reply:
Swift Format is definitely the next thing on my list of "Stuff-I-have-already-worked-on-that-belongs-public" now